src/decode.rs

```diff
@@ -770,9 +769,14 @@ unsafe fn read_pal_plane(
     let used_cache = &used_cache[..i];
 
     // parse new entries
+    #[allow(unused_mut)]
```

Comment:
```suggestion
```

Why's this needed?  You're doing `&mut pal_guard[pli]` so it seems like it would need to be `mut`.